code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -82,7 +82,7 @@
 	name = "accelerator laser cannon"
 	desc = "An advanced laser cannon that does more damage the farther away the target is."
 	icon_state = "lasercannon"
-	inhand_icon_state = "laser"
+	inhand_icon_state = null

```

Comment:
is this one intentional? inhands don't have missing texture icons, so i'm not sure we should have an invisible gun, even if it's sprite isn't unique.

Reply:
Yeah, with the inhands set to null, it allows the proc to change the worn and inhands icons displaying the battery's charge.  Though as you have found, while it fixes and utilizes the existing worn icon when it's on the back, it is invisible while in the hands since those icons don't exist at the moment.

We could set inhands back to laser and let the back icon show missing texture until the unique inhands are created. What's your opinion?